File: benchmarks/raytracer/raytracer-assemblyscript/assembly/camera.ts
import { Vector } from './vector';

/*
 * Class holding data about the camera 
 */
export class Camera {

	constructor(
		// position of the camera
		public position: Vector,
		// coordinates to look at
		public lookat: Vector,
		// field of view (angle in degrees)
		public fov: f32
	) {}

}

File: benchmarks/raytracer/raytracer-rust/src/camera.rs
use crate::vector::Vector;


/*
 * Struct holding data about the camera 
 */
#[derive(Copy, Clone)]
pub struct Camera {
    // position of the camera
    pub position: Vector,
    // coordinates to look at
    pub lookat: Vector,
    // field of view (angle in degrees)
    pub fov: f32,
}